### DISCUSSION TITLE
How to keep up with Duckdb versions?

### DISCUSSION THREAD

**Discussion by solsson** [evidence-dev/evidence#2650] [Q&A] [answered]
I found Evidence really easy to get started with, but I struggle with queries because the enbedded DuckDB i version 0.9.1, according to:

```sql versioninfo
  SELECT version() AS duckdb_version
```

Has anyone found published DucksDB docs for old versions? My searches and experiments using duckdb locally lead to issues similar to evidence-dev/evidence#2157 for example with UNNEST when I try the same in Evidence.

What would it take to lift the version? I notice that latest duckdb-wasm is ["aligned with DuckDB v1.1.1"](https://github.com/duckdb/duckdb-wasm/releases/tag/v1.29.0).

**Comment by solsson** [comment]
If I add to package.json

```
    "duckdb": "1.1.1",
    "duckdb-async": "1.1.1",
    "@duckdb/duckdb-wasm": "1.29.0",
```

then run `npm install`, `rm -r node_modules/@evidence-dev/duckdb/node_modules/duckdb*`, `rm -r node_modules/@evidence-dev/universal-sql/node_modules/@duckdb` and restart evidence, my project still works and `SELECT version() AS duckdb_version` says 1.1.1

npm ls duckdb:

```
my-evidence-project@0.0.1
├─┬ duckdb-async@1.1.1 invalid: "1.0.0" from node_modules/@evidence-dev/duckdb
│ └── duckdb@1.1.1 deduped
└── duckdb@1.1.1
```

npm ls @duckdb/duckdb-wasm

```
my-evidence-project@0.0.1
├── @duckdb/duckdb-wasm@1.29.0
└─┬ @evidence-dev/evidence@39.1.11
  └─┬ @evidence-dev/universal-sql@2.1.5
    └── @duckdb/duckdb-wasm@1.29.0 deduped invalid: "1.28.0" from node_modules/@evidence-dev/universal-sql
```

**Comment by solsson** [comment]
I suspect that the upgrade of @duckdb/duckdb-wasm is the reason my time series are printed as epoch milliseconds instead of dates in BarChart and LineChart.

**Comment by solsson** [comment, marked as answer]
[@evidence-dev/universal-sql@2.2.0](https://github.com/evidence-dev/evidence/releases/tag/%40evidence-dev%2Funiversal-sql%402.2.0) updates duckdb version

**Comment by adam-homeboost** [comment]
A year later and duckdb is now at 1.4.1 and evidence still uses 1.1.1.  Can we get this updated?  Or, better, an easier method to force the version?  Pinning duckdb modules in package.json has no effect.
